examples/llm/inference/convert_composer_to_hf.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2022 MosaicML Examples authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""TBD."""
+
+import json
+import tempfile
+from argparse import ArgumentParser, Namespace
+from pathlib import Path
+from typing import Any, Dict, Optional, Union
+
+import sentencepiece as spm
+import torch
+from composer.utils import get_file, safe_torch_load
+from transformers import (AutoConfig, AutoTokenizer, PretrainedConfig,
+                          PreTrainedTokenizer)
+
+from examples.llm import MosaicGPTConfig
+
+
+def get_hf_config_from_composer_state_dict(
+        state_dict: Dict[str, Any]) -> PretrainedConfig:
+    hf_config_dict = state_dict['state']['integrations']['huggingface'][
+        'model']['config']['content']
+
+    AutoConfig.register('mosaic_gpt', MosaicGPTConfig)
+    hf_config = AutoConfig.for_model(**hf_config_dict)
+    # Always set init_device='cpu'
+    hf_config.init_device = 'cpu'
```

Comment:
modify the config dict before passing it to `for_model` instead of changing the attr after the fact. just a little bit safer, although both should be fine